tensorflow/core/kernels/mkl/mkl_conv_ops.h

```diff
@@ -641,6 +641,9 @@ class MklConvBackpropCommonOp : public OpKernel {
     OP_REQUIRES(context, FormatFromString(data_format_str, &data_format_),
                 errors::InvalidArgument("Invalid data format"));
     OP_REQUIRES_OK(context, context->GetAttr("strides", &strides_));
+    if (strides_.size() < 4) {
+        return errors::InvalidArgument("strides dimensions must not be < 4. ");
+    }
```

Comment:
In mkl_conv_ops.cc, we have the following check for fwd conv op
 747     OP_REQUIRES(context, (strides_.size() == 4 || strides_.size() == 5),
 748                 absl::InvalidArgumentError("Sliding window strides field must "
 749                                            "specify 4 or 5 dimensions"));
If you intend to do it for backward conv ops, why not do the same check? After all, oneDNN conv ops only support 2D and 3D.

Reply:
It's still wrong if the user supplies a strides size of 5 but data only for conv2d.  The  `MklConvCustomBackpropInputOp` will deduce the wrong convolution size.  You need to ensure it is exactly 4 for conv2d, and exactly 5 for conv3d.